### PR TITLE
don't pull image in dockerng.running if test=true

### LINKED ISSUE #36587: dockerng.running downloads images in test=true
### Description of Issue/Question

When I run dockerng.running in test=true it does things instead of saying that it would do things - it seems to download docker images for instead of just saying that it would do so.
Btw why does it download at all? I thought dockerng.image_present does this? Shouldnt dockerng.running fail if the image isnt present?

First run:

```
          ID: rancher_server_container
    Function: dockerng.running
        Name: rancher-server
      Result: None
     Comment: Container 'rancher-server' will be created
     Started: 00:45:26.067037
    Duration: 39270.397 ms
     Changes:   
              ----------
              image:
                  ----------
                  Layers:
                      ----------
                      Pulled:
                          - b171f9dbc13b
                          - ed40d4bcb313
                          - 96c6a1f3c3b0
                          - ccfc4df4fbba
                          - 9e4d5a1238bb
                          - f986ba0d224d
                          - 303d9649bb89
                          - 86a35696481e
                          - 56825bfb5bb7
                          - fabbcc718243
                          - eb2848176ff4
                          - 35489637da84
                          - 5c005e2df0e8
                          - 64ba16c4c9f4
                  Status:
                      Downloaded newer image for rancher/server:stable
                  Time_Elapsed:
                      39.2294640541
```

Second run:

```
          ID: rancher_server_container
    Function: dockerng.running
        Name: rancher-server
      Result: None
     Comment: Container 'rancher-server' will be created
     Started: 00:47:27.581069
    Duration: 10.533 ms
     Changes:  
```
### Setup

I used this state:

/srv/salt/test-container-running-issue.sls:

```
rancher_server_container:
  dockerng.running:
    - name: rancher-server
    - image: rancher/server:stable
    - environment:
      - CATTLE_DB_CATTLE_MYSQL_HOST: test
      - CATTLE_DB_CATTLE_MYSQL_PORT: test
      - CATTLE_DB_CATTLE_MYSQL_NAME: test
      - CATTLE_DB_CATTLE_USERNAME: test
      - CATTLE_DB_CATTLE_PASSWORD: test
    - port_bindings:
      - http://127.0.0.1:8080
    - restart_policy: always
#    - require:
#      - dockerng: rancher_server_image
```
### Steps to Reproduce Issue

salt 'testserver' state.sls 
### Versions Report

Both master and minion are  2016.3.3 (Boron) on Debian 8.

### PR DESCRIPTION
**Pull request by t2b**
### What does this PR do?

prevent state dockerng.unning from actually pulling docker image if run with test=true
### What issues does this PR fix or reference?

fixes saltstack/salt#36587 
### Previous Behavior

dockerng.running pulls the image if it is not present
### New Behavior

don't pull the image if test=true and return an appropriate comment
### Tests written?

No

### CODE REVIEW

**Comment by cachedout** [comment]
@ticosax Any thoughts?

File: salt/states/dockerng.py [hunk 1]
```diff
@@ -1580,6 +1580,13 @@ def running(name,
         image = ':'.join(_get_repo_tag(str(image)))
 
     if image not in __salt__['dockerng.list_tags']():
+        if __opts__['test']:
+            ret['result'] = None
+            ret['comment'] = (
+                'Image {} will be pulled and container {} will be created'
+                .format(image, name)
+            )
+            return ret
```
Comment (ticosax) [inline salt/states/dockerng.py]: I believe you shouldn't return so early, as other actions are expected from `dockerng.running` state once the image is pulled.

**Review by ticosax — CHANGES_REQUESTED** [review]
It is going in the right direction, but it needs more work for completeness.

**Comment by t2b** [comment]
i can return later, but some later actions require a present image or exception will appear. i can try to catch these exceptions and return appropriate comments.

**Comment by ticosax** [comment]
Just do what you think will be useful and not too disruptive for normal operations. I will not be picky with this pull request.

**Comment by t2b** [comment]
i have update my branch. i don't return early now, and the comment is more specific.

if you are running a container 'redis' based on the image 'redis:3.0' and change the state to use the image 'redis:3.2' the comment will be:
`Image 'redis:3.2' will be pulled. Container 'redis' will be replaced`

the new commit includes my changes from pull request #37199

File: salt/states/dockerng.py [hunk 2]
```diff
@@ -1593,7 +1593,10 @@ def running(name,
         else:
             ret['changes']['image'] = pull_result
 
-    image_id = __salt__['dockerng.inspect_image'](image)['Id']
+    try:
+        image_id = __salt__['dockerng.inspect_image'](image)['Id']
+    except CommandExecutionError:
+        image_id = None
```
Comment (ticosax) [inline salt/states/dockerng.py]: here I suggest something like
```python
if not __opts__['test']:
    raise
```
To not break existing behavior

**Review by ticosax — APPROVED** [review]
well done

**Comment by cachedout** [comment]
Thank you @t2b and thanks as always to @ticosax as well.

**Pull request merged**